examples/ContinuumSnakeWithLiftingWaveCase/snake_forcing.py

```diff
@@ -0,0 +1,201 @@
+__doc__ = """ External forces and Actions (in learning) of the snake terrain case."""
+
+import numpy as np
+from elastica._linalg import _batch_matvec
+from elastica.utils import _bspline
+from numba import njit
+from elastica._linalg import (
+    _batch_norm,
+    _batch_product_k_ik_to_ik,
+    _batch_vec_oneD_vec_cross,
+)
+from elastica.external_forces import NoForces
+from elastica.external_forces import (
+    inplace_addition,
+    inplace_substraction,
+)
+
+
+class MuscleTorques_lifting(NoForces):
+    """
+    This class applies muscle torques along the body. The applied muscle torques are treated
+    as applied external forces. This class can apply lifting
+    muscle torques as a traveling wave with a beta spline or only
+    as a traveling wave. For implementation details refer to X. Zhang et. al. Nat. Comm. 2021
+
+            Attributes
+            ----------
+            direction: numpy.ndarray
+                    2D (dim, 1) array containing data with 'float' type. Muscle torque direction.
+            angular_frequency: float
+                    Angular frequency of traveling wave.
+            wave_number: float
+                    Wave number of traveling wave.
+            phase_shift: float
+                    Phase shift of traveling wave.
+            ramp_up_time: float
+                    Applied muscle torques are ramped up until ramp up time.
+            my_spline: numpy.ndarray
+                    1D (blocksize) array containing data with 'float' type. Generated spline.
+            switch_on_time: float
+                    time to switch on the muscle activation.
+    """
+
+    def __init__(
+        self,
+        base_length,
+        b_coeff,
+        period,
+        wave_number,
+        phase_shift,
+        direction,
+        rest_lengths,
+        ramp_up_time=0.0,
+        with_spline=False,
+        switch_on_time=0.0,
+        *args,
+        **kwargs,
+    ):
+        """
+
+        Parameters
+        ----------
+        base_length: float
+                Rest length of the rod-like object.
+        b_coeff: nump.ndarray
+                1D array containing data with 'float' type.
+                Beta coefficients for beta-spline.
+        period: float
+                Period of traveling wave.
+        wave_number: float
+                Wave number of traveling wave.
+        phase_shift: float
+                Phase shift of traveling wave.
+        direction: numpy.ndarray
+           1D (dim) array containing data with 'float' type.
+        ramp_up_time: float
+                Applied muscle torques are ramped up until ramp up time.
+        with_spline: boolean
+                Option to use beta-spline.
+        switch_on_time: float
+                time to switch on the muscle activation.
+
+        """
+        super(MuscleTorques_lifting, self).__init__()
```

Comment:
```suggestion
       super().__init__()
```

https://peps.python.org/pep-3135/

Reply:
Changed in both MuscleTorquesLifting and SnakeRodPlaneContact